### PR TITLE
PDA starts in ID slot

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

Say goodbye to this minor inconvenience, and hours of my life.
The PDA will now start in the ID slot with the ID already inserted. Works with most playable jobs. Has been tested. Must be enabled in the game preferences.

# Wiki Documentation

No changes needed.

# Changelog

:cl:  
tweak: pda now starts in the id slot with the id inserted for most playable jobs.
tweak: adds an option to the game preferences for whether you want this change or not.
/:cl:

### CODE REVIEW

**Comment by KillerOrcaCora** [comment]
Good Pr you are an angel.

**Comment by patpol4** [comment]
Hugbox

**Comment by ghost** [comment]
Two clicks saved every round. That shit adds up.

**Comment by JamieD1** [comment]
Make it a preference, can help with it.

**Comment by bruhlookatthisdood** [comment]
halelughija

**Comment by JamieD1** [comment]
You are forcing the ID slot to have the PDA on almost all the jobs, It would be alot easier if you just left it all as default and then ran the slot id into pda if the preference is there and then move the pda to the id slot.

**Comment by JamieD1** [comment]
Everyone with this enabled got an ID in a PDA, however no ID if the pref is disabled.

Please test and try again.

**Comment by adamsong** [comment]
No one spawned with ids or pda

File: code/modules/jobs/job_types/_job.dm [hunk 1]
```diff
@@ -286,14 +287,27 @@
...
-	var/obj/item/pda/PDA = H.get_item_by_slot(pda_slot)
+	var/obj/item/pda/PDA = new pda_type()
 	if(istype(PDA))
 		PDA.owner = H.real_name
 		if(H.mind?.role_alt_title)
 			PDA.ownjob = H.mind.role_alt_title
 		else
 			PDA.ownjob = J.title
+
+		if (H.client.prefs.id_in_pda)
```
Comment (nmajask) [inline code/modules/jobs/job_types/_job.dm]: This won't work, clients are transferred after the client is equipped on round-start characters due to some messed up code. You can sidestep this by adding a var to the humans and assigning the value of that var for player's character equal to their preference. Basically, copy what skirt prefs did, as that works.

**Review by nmajask — CHANGES_REQUESTED** [review]

**Review by nmajask — CHANGES_REQUESTED** [review]
Revert the changes to yarn

**Comment by SapphicOverload** [comment]
> No one spawned with ids or pda

@adamsong i fixed it

**Comment by ToasterBiome** [comment]
Mining medic was missing their ID, fixed I think

**Review by nmajask — DISMISSED** [review]
Revert the changes to the horror files

**Comment by ToasterBiome** [comment]
Haven't seen any other bugs with it, @JamieD1 ready to merge?

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
For some reason, IDs don't start with their modular access ie department engineers dont have access to their department, might be something about this PR

**Review by nmajask — COMMENTED** [review]
In the security officer and station engineer files on lines 109 and 82 respectively, you can either use GetID() on the item in the wear_id slot or use get_idcard() on the mob to get the actual ID which would fix ToasterBiome's issue

**Review by JamieD1 — APPROVED** [review]

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -206,13 +206,14 @@
...
-	id = /obj/item/card/id
 	ears = /obj/item/radio/headset
-	belt = /obj/item/pda
 	back = /obj/item/storage/backpack
 	shoes = /obj/item/clothing/shoes/sneakers/black
 	box = /obj/item/storage/box/survival
 
+	var/obj/item/id_type = /obj/item/card/id
+	var/obj/item/pda_type = /obj/item/pda
+
```
Comment (JamieD1) [inline code/modules/jobs/job_types/_job.dm]:
```suggestion
```

**Review by ToasterBiome — DISMISSED** [review]
People show up twice on PDA sometimes and I think this has something to do with it, will be looking in the coming days for anyone with dual-pda syndrome.

**Comment by JamieD1** [comment]
> People show up twice on PDA sometimes and I think this has something to do with it, will be looking in the coming days for anyone with dual-pda syndrome.

Find anything?

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**